src/Core/Framework/Bundle.php

```diff
@@ -115,17 +115,18 @@ protected function getServicesFilePath(): string
         return 'Resources/config/services.xml';
     }
 
-    protected function registerFilesystem(ContainerBuilder $container, string $key): void
+    protected function registerFilesystem(ContainerBuilder $container, string $key, ?string $baseKey = null): void
     {
         $containerPrefix = $this->getContainerPrefix();
-        $parameterKey = sprintf('shopware.filesystem.%s', $key);
+        $parameterKey = sprintf('shopware.filesystem.%s', $baseKey ?? $key);
         $serviceId = sprintf('%s.filesystem.%s', $containerPrefix, $key);
+        $suffix = empty($baseKey) ? '' : '-' . $key;
```

Comment:
should this not be separated by `/` to get different directories in `plugins/my_plugin/my_custom_filesystem`?

Reply:
I think this is up to debate but here are my thoughts on this:
The `$containerPrefix` is lower camel case. So if I have plugin `FooBar` it gets the folder `foo_bar`. When I have the plugin `FooBarParty` I get `foo_bar_party`. But when I separate the subdirectories by `_` then the private directory of `foo_bar_party` is ambiguous to FooBars private directory `party` as both resolve to `foo_bar_party`.
Why do I add multiple directories next to each other? Like this I do not break existing logic by expecting certain folders and have no overlapping file systems.
And I definitely do not separate by `/` but always `DIRECTORY_SEPARATOR` but this is content of an other future PR.

Reply:
so you want to have directories like this?
```
files/plugins/my_plugin-party/party.png
files/plugins/my_plugin-foo/foo.txt
files/plugins/my_plugin/test.txt
```

Reply:
This is the current suggestion. Those file systems neither overlap nor are ambiguous nor a breaking change. But this is debatable as long as the same criteria are given.